### ISSUE TITLE
Ntlmrelayx doesn't relay or log connections when -tf is used 

### ISSUE THREAD

**Issue by Term1N8** [fortra/impacket#1676]
### Configuration  
impacket version:  v0.11.0
Python version:  3.8.10
Target OS:  Ubuntu 20.04.6

Command: ntlmrelayx.py -tf hosts.txt -smb2support -debug -of hashes.txt

Result: no connections are logged, no debug activity, and hashes are logged in hashes.txt. When replacing -tf with -t <IP HERE> connections are received/logged as normal. Tried using a targets file generated by netexec (new crackmapexec), as well as manually.

**Comment by PocketDC** [comment]
Same problem here,` -tf` gives me nothing but individual targets (from the same file) will

**Comment by Term1N8** [comment]
The problem above occurred when installing via pipx. When using the ntlmrelayx.py file in the examples dir, a different issue happens where every connection says "connection will be relayed after re-authentication". Works fine with -t "host", as soon as I switch to -tf list.txt it breaks.

**Comment by Zamanry** [comment]
My team had multiple unique hosts within a file and we had successful execution of ntlmrelay. Python 3.9 on Kali Linux. We installed Impacket via apt-get. Impacket v0.10.0

**Comment by mbb5546** [comment]
Also having these issues, I Installed impacket by cloning the project and then configuring a venv with virtualenv. From within the venv I ran the setup.py script.

Running ntlmrelayx with the `-tf` flag seems broken and when relaying authentication the following message is received over and over again 

[*] Received connection from ACME/User123 at Computerabc, connection will be relayed after re-authentication

Pivoting to using the `-t` flag works as normal. `-tf` seems to be working incorrectly.

**Comment by gabrielg5** [comment]
Hey hello,

Going through the ticket to understand this scenario and relate with latest changes in ntlmrelayx. It seems to be related with the multi-relay feature (#767).

Multi-relay is enabled when using `-tf` and disabled when using `-t` by default. That's why you faced different behaviors when defining targets one way or the other.

Multirelay leverages client's ability to reauthenticate after triggering a first authentication request. So, if client does not honour this condition, no connection will end up being relayed ([We Love Relaying Credentials: A Technical Guide to Relaying Credentials Everywhere (secureauth.com)](https://www.secureauth.com/resources/we-love-relaying-credentials-a-technical-guide-to-relaying-credentials-everywhere/#header4:~:text=How%20to%20attack%20II.%20The%20Multi%2Drelay%20feature))

A couple months ago we merged a PR (#1602) that added a log message when first connection attempt was received in the relay server; letting users know that an authentication was performed - even if, by how the client behaves, it is not relayed afterwards -
That's the behavior change @mbb5546 comments about; that's the log message that was added.

Multi-relay capabilities can be disabled with the flag `--no-multirelay` (#1128). Useful, among others, in scenarios like these you are facing now

**Comment by gabrielg5** [comment]
Hi @kopfjager007,
That seems like another issue. Will try to replay what you mention in my lab related to the `-w` flag.

In the meantime, have you tried with the newly added `--keep-relaying` flag? We added it a couple weeks ago in the context of #1741
I guess it will come in handy in your scenario as you need to keep considering targets despite having a relayed connection

Let us know how that goes.

**Comment by kopfjager007** [comment]
> Hi @kopfjager007, That seems like another issue. Will try to replay what you mention in my lab related to the `-w` flag.
> 
> In the meantime, have you tried with the newly added `--keep-relaying` flag? We added it a couple weeks ago in the context of #1741 I guess it will come in handy in your scenario as you need to keep considering targets despite having a relayed connection
> 
> Let us know how that goes.

Thanks for the reply @gabrielg5! I saw the new flag `--keep-relaying` after (naturally...) I posted the previous comment. Using `--keep-relaying` performs  as expected and "solves" my issue. Although the real issue for me turns out to be me not reading, so I'll just go sit in the corner for a while for that :). Thanks again!

**Comment by gabrielg5** [comment]
Closed after #1789

thank you all

**Comment by vpaschalidis** [comment]
I still have the issue. The CONTOSO/Administrator has admin access and can connect remotely to the targets.

scenario 1:

impacket-ntlmrelayx -tf smb_targets.txt -smb2support -socks --output-file ntlm_relay_hashes.txt


ntlmrelayx> [*] Received connection from CONTOSO/Administrator at LAN-SQL01, connection will be relayed after re-authentication
[*] Received connection from CONTOSO/Administrator at LAN-SQL01, connection will be relayed after re-authentication
[*] Received connection from CONTOSO/Administrator at LAN-SQL01, connection will be relayed after re-authentication
[*] Received connection from CONTOSO/Administrator at LAN-SQL01, connection will be relayed after re-authentication
[*] Received connection from CONTOSO/Administrator at LAN-SQL01, connection will be relayed after re-authentication

scenario 2:

impacket-ntlmrelayx -t 10.0.0.50 -socks -smb2support --output-file ntlm_relay_hashes.txt

ntlmrelayx> [*] SMBD-Thread-9 (process_request_thread): Received connection from 10.0.0.12, attacking target smb://10.0.0.50
[-] Authenticating against smb://10.0.0.50 as CONTOSO/ADMINISTRATOR FAILED
[*] All targets processed!
[*] SMBD-Thread-10 (process_request_thread): Connection from 10.0.0.12 controlled, but there are no more targets left!
[*] All targets processed!
[*] SMBD-Thread-11 (process_request_thread): Connection from 10.0.0.12 controlled, but there are no more targets left!
